pkg/reachability/reachability.go

```diff
@@ -65,7 +84,34 @@ func PerformReachabilityAnalysis(enabled bool, r reporter.Reporter, purls []stri
 
 			switch filepath.Ext(d.Name()) {
 			case ".java":
-				err = javaReachabilityDetector.Detect(dir, path, detectionResults, advisoriesToCheckPerLanguage["java"])
+				g.Go(func() error {
+					// Get a detector from the pool
+					detector := <-detectorPool
```

Comment:
What happens if there's nothing available in the pool? After creating the errgroup I think we should be setting a limit on the number of goroutines it can spawn with `g.SetLimit(workerCount)`